dev/com.ibm.ws.logging/src/com/ibm/ws/logging/internal/impl/BaseTraceFormatter.java

```diff
@@ -1001,6 +996,53 @@ public String formatObj(Object objs) {
         return ans;
     }
 
+    private String formatObject(Object objs, int depth) {
+        if (objs == null) {
+            return null;
+        }
+        if (depth >= COLLECTION_DEPTH_LIMIT) {
+            return "Collection depth limit exceeded";
+        }
+        if (objs instanceof Collection) {
+            try {
+                Iterator<?> iter = ((Collection<?>) objs).iterator();
+                StringBuilder sb = new StringBuilder("[");
+                if (iter.hasNext()) {
+                    sb.append(formatObject(iter.next(), depth + 1));
+                }
+                while (iter.hasNext()) {
+                    sb.append(", ").append(formatObject(iter.next(), depth + 1));
+                }
+                sb.append("]");
+                return sb.toString();
+            } catch (Throwable t) {
+                return "[Caught " + t.toString() + " while logging collection type " + objs.getClass().getName() + "]";
+            }
+        }
+        if (objs instanceof Untraceable) {
+            return objs.getClass().getName(); // Use only the class name of the object
+        } else if (objs instanceof Traceable) {
+            return formatTraceable((Traceable) objs);
+        } else if (objs instanceof TruncatableThrowable) {
+            return DataFormatHelper.throwableToString((TruncatableThrowable) objs);
+        } else if (objs instanceof Throwable) {
+            return DataFormatHelper.throwableToString(new TruncatableThrowable((Throwable) objs));
+        }
+        try { // Protect ourselves from badly behaved toString methods
+            Class<?> cls = objs.getClass();
+            String className = cls.getName();
+            if (Proxy.isProxyClass(cls) || className.contains("$Proxy$_$$_Weld")) {
+                return "Proxy for " + className;
```

Comment:
This String does not make sense.  It does not give you the class of the object that is being proxied, so it would make more sense to me for it to just do the default toString() behavior:
```
className + "@" + Integer.toHexString(System.identityHashCode(objs))
```

Reply:
done